src/uproot/models/TTree.py

```diff
@@ -905,3 +911,161 @@ def read_members(self, chunk, cursor, context, file):
+        parse_colon = False
+        files = files2
+
+    if uproot._util.isstr(files):
+        if parse_colon:
+            file_path, object_path = uproot._util.file_object_path_split(files)
+        else:
+            file_path, object_path = files, None
+
+        parsed_url = urlparse(file_path)
+
+        if parsed_url.scheme.upper() in uproot._util._remote_schemes:
+            yield file_path, object_path
+
+        else:
+            expanded = os.path.expanduser(file_path)
+            if _regularize_files_isglob.search(expanded) is None:
+                yield file_path, object_path
+
+            else:
+                matches = list(_regularize_files_braces.finditer(expanded))
+                if len(matches) == 0:
+                    results = [expanded]
+                else:
+                    results = []
+                    for combination in itertools.product(
+                        *[match.group(0)[1:-1].split(",") for match in matches]
+                    ):
+                        tmp = expanded
+                        for c, m in list(zip(combination, matches))[::-1]:
+                            tmp = tmp[: m.span()[0]] + c + tmp[m.span()[1] :]
+                        results.append(tmp)
+
+                seen = set()
+                for result in results:
+                    for match in glob.glob(result):
+                        if match not in seen:
+                            yield match, object_path
+                            seen.add(match)
+
+    elif isinstance(files, uproot.behaviors.TBranch.HasBranches):
+        yield files, None
+
+    elif isinstance(files, dict):
+        for key, object_path in files.items():
+            for file_path, _ in _regularize_files_inner(key, False, counter):
+                yield file_path, object_path
+
+    elif isinstance(files, Iterable):
+        for file in files:
+            counter[0] += 1
+            for file_path, object_path in _regularize_files_inner(
+                file, parse_colon, counter
+            ):
+                yield file_path, object_path
+
+    else:
+        raise TypeError(
+            "'files' must be a file path/URL (string or Path), possibly with "
+            "a glob pattern (for local files), a dict of "
+            "{{path/URL: TTree/TBranch name}}, actual TTree/TBranch objects, or "
+            "an iterable of such things, not {0}".format(repr(files))
+        )
+
+
+def _regularize_files(files):
+    out = {}
+    seen = set()
+    counter = [0]
+    for file_path, object_path in _regularize_files_inner(files, True, counter):
+        if uproot._util.isstr(file_path):
+            key = (counter[0], file_path, object_path)
+            if key not in seen:
+                out[file_path] = object_path
+                seen.add(key)
+        else:
+            out[file_path] = object_path
+
+    if len(out) == 0:
+        raise uproot._util._file_not_found(files)
+
+    return out
```

Comment:
It looks like `_regularize_files` and its helper function are (still) [also defined in TBranch.py](https://github.com/scikit-hep/uproot4/blob/7fc2b39bc208fed8fda12233279f2b166e9b9977/src/uproot/behaviors/TBranch.py#L2987-L3119).

There should only be one definition of this function, and since both behaviors/TBranch.py and models/TTree.py want access to it, it should be in a neutral location: _util.py. I also selected `_regularize_object_path` because that should go along wherever `_regularize_files` goes.

Actually, a lot of the bulk of TBranch.py is helper functions, and many of them ought to be in _util.py. But let's just focus on these three (`_regularize_files`, its helper function, and `_regularize_object_path`).